src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/HMPPSAuthService.kt

```diff
@@ -0,0 +1,61 @@
+package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.service
+
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.http.HttpStatus
+import org.springframework.stereotype.Service
+import org.springframework.web.reactive.function.client.WebClient
+import org.springframework.web.reactive.function.client.WebClientResponseException
+import org.springframework.web.util.UriComponentsBuilder
+import reactor.core.publisher.Flux
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.AuthGroupID
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.AuthUser
+
+val AuthServiceProviderGroupPrefix = "INT_SP_"
+
+data class AuthGroup(
+  val groupCode: String,
+  val groupName: String,
+)
+
+@Service
+class HMPPSAuthService(
+  @Value("\${hmppsauth.api.locations.user-groups}") private val userGroupsLocation: String,
+  private val hmppsAuthApiWebClient: WebClient,
+) {
+  fun getServiceProviderOrganizationForUser(user: AuthUser): AuthGroupID? {
+    val url = UriComponentsBuilder.fromPath(userGroupsLocation)
+      .buildAndExpand(user.userName)
+      .toString()
+
+    val groups = hmppsAuthApiWebClient.get().uri(url)
+      .retrieve()
+      .bodyToFlux(AuthGroup::class.java)
+      .onErrorResume { e ->
+        when (e) {
+          is WebClientResponseException -> {
+            // we expect 404s for non-auth users
+            if (!e.statusCode.equals(HttpStatus.NOT_FOUND)) {
+              log.error("could not get groups for user", e)
+            }
+          }
+          else -> log.error("could not get groups for user", e)
+        }
+        Flux.empty()
+      }
+      .collectList().block()
```

Comment:
Is there a reason we are blocking here rather than using subscribe?

Reply:
yeh, we need the result right away